Mathlib/Topology/ContinuousMap/Basic.lean

```diff
@@ -50,21 +50,24 @@ protected theorem continuousAt (f : C(α, β)) (x : α) : ContinuousAt f x :=
 theorem map_specializes (f : C(α, β)) {x y : α} (h : x ⤳ y) : f x ⤳ f y :=
   h.map f.2
 
-section
-
-variable (α β)
+section DiscreteTopology
+variable [DiscreteTopology α]
 
+variable (α β) in
 /--
 The continuous functions from `α` to `β` are the same as the plain functions when `α` is discrete.
 -/
 @[simps]
-def equivFnOfDiscrete [DiscreteTopology α] : C(α, β) ≃ (α → β) :=
+def equivFnOfDiscrete : C(α, β) ≃ (α → β) :=
   ⟨fun f => f,
     fun f => ⟨f, continuous_of_discreteTopology⟩,
     fun _ => by ext; rfl,
     fun _ => by ext; rfl⟩
 
-end
+@[simp] lemma coe_equivFnOfDiscrete (f : C(α, β)) : equivFnOfDiscrete α β f = f := rfl
```

Comment:
I'd expect the coe_ lemma to be `equivFnOfDiscrete α β = id`. And probably this one to be `equivFnOfDiscrete_apply` (although `coe_` in front here would probably be fine too)

Reply:
Looking at other equiv lemmas, this seems the general pattern: see `Equiv.coe_refl` and `Equiv.refl_apply` for a trivial example, and similar for `trans`

Reply:
> I'd expect the coe_ lemma to be `equivFnOfDiscrete α β = id`.

Your expectation doesn't typecheck 😉

> Looking at other equiv lemmas, this seems the general pattern: see `Equiv.coe_refl` and `Equiv.refl_apply` for a trivial example, and similar for `trans`

This doesn't really apply here, because we're talking about a function `φ : (X → Y) →bundled₁ X →bundled₂ Y` that's twice bundled. So there could be three lemmas:
1. The coercion of `φ` to `(X → Y) → X →bundled₂ Y`
2. The coercion of `φ f` to `X → Y`
3. The value of `φ f x` in `Y`

This PR currently has
1. Nothing, because I doubt people care about this lemma
2. `coe_`
3. `_apply`

You are arguing for
1. `coe_`
2. `_apply`
3. Unclear to me

which doesn't necessarily strike me as better

Reply:
In what sense is this function twice bundled? The declaration we're talking about has `C(α, β) ≃ (α → β)`, which is only singly bundled. I can see plenty of examples of the pattern I'm asking for: it's better because it's how every other bundled coercion lemma is stated!

Reply:
And if you're talking about the inverse of this equivalence, this PR currently has 
2. `coe_`
3. `_apply_apply`
whereas I'm arguing for
1. `coe_`
2. `_apply`
3. `_apply_apply`
which is consistent with how other twice bundled functions are named

Reply:
Okay, we now have for the forward direction
1. `coe_`
2. Nothing
3. `_apply`

and for the backward direction
1. `coe_`
2. `_apply`
3. `_apply_apply`

I genuinely do not think this is an improvement over what I had originally. But I also do not care. This is incredibly minor and I have better things to do.